__tests__/environment-test.js

```diff
@@ -0,0 +1,81 @@
+"use strict"
+
+jest.dontMock('../environment.js');
+
+describe('environment', () => {
+  let environment = require('../environment.js');
+
+  it('uses original config attributes if no env key is found', () => {
```

Comment:
I'm really thankful for all the tests you added but in this case, I don't think that they are a good trade-off.

- With my suggested change, the logic in the function now becomes a one liner that is unlikely going to get a lot of issues.
- I don't really want to extract that logic out into its own separate file for now.
- The tests you wrote are unlikely to catch bugs and probably going to cause more work to keep them updated (see more inline)